packages/instant/webpack.config.js

```diff
@@ -11,22 +12,107 @@ const GIT_SHA = childProcess
     }
 
     return undefined;
 };
 
+const ROLLBAR_PUBLISH_TOKEN_ENV_NAME = 'INSTANT_ROLLBAR_PUBLISH_TOKEN';
+const ROLLBAR_CLIENT_TOKEN_ENV_NAME = 'INSTANT_ROLLBAR_CLIENT_TOKEN';
+const getRollbarPlugin = environmentName => {
+    if (!environmentName) {
+        return undefined;
+    }
+
+    const publishToken = process.env[ROLLBAR_PUBLISH_TOKEN_ENV_NAME];
+    if (!publishToken) {
+        return undefined;
+    }
+
+    let rollbarPublicPath;
+    if (environmentName === 'dogfood') {
+        rollbarPublicPath = 'http://0x-instant-dogfood.s3-website-us-east-1.amazonaws.com';
+    } else if (environmentName === 'staging') {
+        rollbarPublicPath = 'http://0x-instant-staging.s3-website-us-east-1.amazonaws.com';
+    } // TODO(sk): When we decide on JS cdn, add public path here
+
+    if (!rollbarPublicPath) {
+        console.log('No rollbar public path');
+        return undefined;
+    }
+
+    const rollbarPluginOptions = {
+        accessToken: publishToken,
+        version: GIT_SHA,
+        publicPath: rollbarPublicPath,
+    };
+    return new RollbarSourceMapPlugin(rollbarPluginOptions);
+};
+
+const validateRollbar = (environmentName, rollbarPlugin) => {
+    const requiresRollbar = environmentName === 'dogfood' || environmentName === 'staging';
+
+    if (!requiresRollbar) {
+        return;
+    }
+
+    if (!process.env[ROLLBAR_CLIENT_TOKEN_ENV_NAME]) {
+        throw new Error(`${ROLLBAR_CLIENT_TOKEN_ENV_NAME} must be set for ${environmentName}`);
+    }
+
+    if (!rollbarPlugin) {
+        if (environmentName === 'dogfood' || environmentName === 'staging') {
+            throw new Error(
+                `Please set rollbar env var ${ROLLBAR_PUBLISH_TOKEN_ENV_NAME} to a Rollbar project access token with post_server_item permissions to deploy source maps to ${environmentName}`,
+            );
+        }
+    }
+};
+
 module.exports = (env, argv) => {
+    const environmentName = getEnvironmentName(env, argv);
     const outputPath = process.env.WEBPACK_OUTPUT_PATH || 'umd';
+
+    let plugins = [
+        new webpack.DefinePlugin({
+            'process.env': {
+                GIT_SHA: JSON.stringify(GIT_SHA),
+                NPM_PACKAGE_VERSION: JSON.stringify(process.env.npm_package_version),
+                HEAP_ANALYTICS_ID: getHeapAnalyticsId(environmentName),
+                ROLLBAR_ENVIRONMENT: JSON.stringify(environmentName),
+                ROLLBAR_CLIENT_TOKEN: JSON.stringify(process.env[ROLLBAR_CLIENT_TOKEN_ENV_NAME]),
+                ROLLBAR_FORCE_DEVELOPMENT_REPORT: JSON.stringify(process.env.INSTANT_ROLLBAR_FORCE_DEVELOPMENT_REPORT),
+            },
+        }),
+    ];
+
+    const rollbarPlugin = getRollbarPlugin(environmentName);
+    if (rollbarPlugin) {
+        console.log('Using rollbar plugin');
+        plugins = plugins.concat(rollbarPlugin);
+    } else {
+        console.log('Not using rollbar plugin');
+    }
+    validateRollbar(environmentName, rollbarPlugin);
```

Comment:
Should we validate it if we are not using it?

Reply:
This function will throw an error if we are an env that requires sourcemaps, but there is no `rollbarPlugin` set.  Perhaps renaming the function to something like `validateRollbarPresence` may make things more clear.